### PR TITLE
Support / use TLID in core tracklist calls

### PR DESCRIPTION
**Pull request by adamcik**

### CODE REVIEW

File: mopidy/core/tracklist.py [hunk 1]
```diff
@@ -200,19 +200,52 @@ def set_single(self, value):
...
         """
         The position of the given track in the tracklist.
 
+        If neither *tl_track* or *tlid* is given we return the index of
+        the currently playing track.
+
         :param tl_track: the track to find the index of
         :type tl_track: :class:`mopidy.models.TlTrack` or :class:`None`
+        :param tlid: of the track to find the index of
+        :type tlid: TLID number or :class:`None`
```
Comment (jodal) [inline mopidy/core/tracklist.py]: s/of the/TLID of the/
s/TLID number/int/

File: mopidy/core/tracklist.py [hunk 2]
```diff
@@ -301,7 +369,9 @@ def previous_track(self, tl_track):
         if position in (None, 0):
             return None
 
-        return self.get_tl_tracks()[position - 1]
+        # Note that since we know we are at position 1-n we know this will
```
Comment (jodal) [inline mopidy/core/tracklist.py]: s/Note that since/Since/
s/1-n/n-1/ ?

File: mopidy/core/tracklist.py [hunk 3]
```diff
@@ -200,19 +200,52 @@ def set_single(self, value):
...
+            for i, tl_track in enumerate(self._tl_tracks):
+                if tl_track.tlid == tlid:
+                    return i
+        return None
+
+    def get_eot_tlid(self):
+        """
+        The TLID of the track that will be played after the given track.
+
+        Not necessarily the same track as :meth:`get_next_tlid`.
```
Comment (jodal) [inline mopidy/core/tracklist.py]: s/track as/TLID as returned by/

File: mopidy/core/tracklist.py [hunk 4]
```diff
@@ -200,19 +200,52 @@ def set_single(self, value):
...
+
+        :rtype: TLID or :class:`None`
```
Comment (jodal) [inline mopidy/core/tracklist.py]: s/TLID/int/

File: mopidy/utils/deprecation.py [hunk 5]
```diff
@@ -45,13 +45,25 @@
     'core.tracklist.remove:kwargs_criteria':
         'tracklist.remove() with "kwargs" as criteria is deprecated',
 
+    'core.tracklist.eot_track':
+        'tracklist.eot_track() is deprecated, use tracklist.get_eot_tlid()',
```
Comment (jodal) [inline mopidy/utils/deprecation.py]: s/deprecated/pending deprecation/ ?

**Comment by jodal** [comment]
+1 with comments

**Comment by adamcik** [comment]
Fixed

File: mopidy/core/tracklist.py [hunk 6]
```diff
@@ -235,6 +269,23 @@ def eot_track(self, tl_track):
...
+        """
+        The tlid of the track that will be played if calling
+        :meth:`mopidy.core.PlaybackController.next()`.
+
+        For normal playback this is the next track in the tracklist. If repeat
+        is enabled the next track can loop around the tracklist. When random is
+        enabled this should be a random track, all tracks should be played once
+        before the tracklist repeats.
+
+        :rtype: TLID or :class:`None`
```
Comment (jodal) [inline mopidy/core/tracklist.py]: s/TLID/int/

File: mopidy/core/tracklist.py [hunk 7]
```diff
@@ -249,35 +300,51 @@ def next_track(self, tl_track):
...
+    def get_previous_tlid(self):
+        """
+        Returns the TLID of the  track that will be played if calling
+        :meth:`mopidy.core.PlaybackController.previous()`.
+
+        For normal playback this is the previous track in the tracklist. If
+        random and/or consume is enabled it should return the current track
+        instead.
+
+        :rtype: TLID or :class:`None`
```
Comment (jodal) [inline mopidy/core/tracklist.py]: s/TLID/int/

**Comment by jodal** [comment]
Since you didn't fix all occurrences of s/TLID/int/, I added comments on the two I can see left. :-)

**Comment by adamcik** [comment]
There.

**Pull request merged**